tzrec/datasets/dataset.py

```diff
@@ -317,6 +321,92 @@ def _build_batch(self, input_data: Dict[str, pa.Array]) -> Batch:
                 input_data = _expand_tdm_sample(
                     input_data, pos_sampled, neg_sampled, self._data_config
                 )
+            elif self._enable_hstu:
+                seq_attr = self._sampler._item_id_field
+                if pa.types.is_string(input_data[seq_attr].type):
```

Comment:
Make line 326-376 a function and move it to datasets/utils.py. Add comments to explain the logic and add a unit test for the function.